### PR TITLE
fix number inputs (fixes #48)

### LINKED ISSUE #48: Does not apply transforms to values typed in
When entering transform values by typing (using numeric keys), the entity does not take on these changes.

Seems to work fine when you click and drag the numbers.

### PR DESCRIPTION
**Pull request by ngokevin**
- don't use refs

### CODE REVIEW

File: src/components/widgets/Vec3Widget.js [hunk 1]
```diff
@@ -6,26 +6,33 @@ import NumberWidget from './NumberWidget';
...
-    this.state = {value: props.value};
-  }
-
-  getValue = () => {
-    return {x: this.refs.x.state.value, y: this.refs.y.state.value, z: this.refs.z.state.value};
+    this.state = {
+      x: this.props.value.x,
+      y: this.props.value.y,
+      z: this.props.value.z
+    }
```
Comment (cvan) [inline src/components/widgets/Vec3Widget.js]: `;`

File: src/components/widgets/Vec3Widget.js [hunk 2]
```diff
@@ -6,26 +6,33 @@ import NumberWidget from './NumberWidget';
...
   }
 
   onChange = (entity, componentName, name, value) => {
-    if (this.props.onChange) {
-      value = this.getValue();
-      this.props.onChange(entity, componentName, name, value);
-    }
+    this.setState({[name]: value}, () => {
```
Comment (cvan) [inline src/components/widgets/Vec3Widget.js:23]: forgive my ignorance - what's this do exactly?

**Comment by fernandojsg** [comment]
It breaks the keyboard input of the numbers (when you click on a number), as when you start typing it will apply the `toFixed` and won't let you finish entering your number

File: src/components/widgets/Vec3Widget.js (hunk 2 above)
Comment (ngokevin) [inline src/components/widgets/Vec3Widget.js:23]: Computed property names: https://github.com/lukehoban/es6features#enhanced-object-literals

**Comment by fernandojsg** [comment]
Thank you!, it also fixes https://github.com/aframevr/aframe-editor/issues/47

**Comment by ngokevin** [comment]
Oh wow, didn't know it fixed that too!

**Pull request merged**